### PR TITLE
Fix #249: Hash in links in --lazy mode

### LINKED ISSUE #249: Hash in links doesn't work in --lazy mode
[Issuehunt badges](https://issuehunt.io/r/saberland/saber/issues/249)

- [x] I confirm that this is an issue rather than a question.

#### Version

v0.6.8

#### Details

屏幕快照_2019-06-09_下午6_44_38

It will be redirected to `/docs/saber-config.html` instead of `/docs/saber-config.html#options`.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $100.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($100.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/saberland/saber/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by lubien**
**Summary**

Closes #249 

This implementation encodes links with # that the user navigates
to then webpack properly decodes into a file valid path. That means:

1. User clicks in "/path/to/file.html#some-id".
2. Webpack receives req.query.route exactly that way and convert.
3. The variable pathname constains a decoded "/path/to/file.html".
4. Do it's proper checks and compile if needed.
5. Return req.query.route (the original link) to the user.

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI/CSS related code, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [x] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: packages/saber/vue-renderer/lib/index.js [hunk 1]
```diff
@@ -403,18 +403,18 @@ class VueRenderer {
...
 
     server.get('/_saber/visit-page', async (req, res) => {
-      const pathname = removeTrailingSlash(decodeURI(req.query.route))
+      const pathname = normalizeRoutePath(req.query.route)
       log.info(`Navigating to ${pathname}`)
       res.end()
 
       if (this.builtRoutes.has(pathname)) {
-        hotMiddleware.publish({ action: 'router:push', route: pathname })
+        hotMiddleware.publish({ action: 'router:push', route: req.query.route })
```
Comment (egoist) [inline packages/saber/vue-renderer/lib/index.js]: We need to ensure there's no trailing hash in `req.query.route`

Comment (lubien) [reply packages/saber/vue-renderer/lib/index.js]: Got it. 

https://github.com/saberland/saber/pull/270/commits/52ac92448940b3a9dd9dd648d35a23675c673238

Comment (egoist) [reply packages/saber/vue-renderer/lib/index.js]: oops, it's a typo, I mean `slash` 😆

Comment (lubien) [reply packages/saber/vue-renderer/lib/index.js]: Oh. 

But also I kinda liked how you refactored it. I definitely learnt something after just waking up 😉

**Pull request merged**